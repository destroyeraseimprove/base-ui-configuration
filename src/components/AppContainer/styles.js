import globalStyles from "styles/global.styles.js";

export default {
  ...{ ...globalStyles },

  appContainer: {
    background: "yellow",
    padding: 5,
  },
};
