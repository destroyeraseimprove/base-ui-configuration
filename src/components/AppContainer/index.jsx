import React, { useState, useRef, useEffect } from "react";
import { createUseStyles } from "react-jss";
import styles from "./styles";
import TestComponent1 from "components/TestComponent1";

export default () => {
  const classes = createUseStyles(styles)();

  return (
    <div className={`${classes.appContainer} roboto`}>
      App Content
      <TestComponent1 />
    </div>
  );
};
