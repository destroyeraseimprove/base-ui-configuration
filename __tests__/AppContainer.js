import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import AppContainer from '../src/components/AppContainer/index.jsx'

describe('AppContainer Component', ()=>{
    it('AppContainer snapshot', ()=>{
        const appContainer = render(<AppContainer/>);
        expect(appContainer).toMatchSnapshot();
    });
    it('renders successfully', ()=>{
        render(<AppContainer/>);
        expect(screen.getByText(/App Content/)).toBeInTheDocument();
    });
});