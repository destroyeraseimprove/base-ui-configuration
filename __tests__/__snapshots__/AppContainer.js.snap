// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer Component AppContainer snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="appContainer-0-2-1 roboto"
      >
        App Content
        <div
          class="testDataWrapper-0-2-2"
        >
          <h5>
            Test data received from http server using AJAX request
          </h5>
          <div
            class="loader"
          >
            <div
              class="loader_base"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="appContainer-0-2-1 roboto"
    >
      App Content
      <div
        class="testDataWrapper-0-2-2"
      >
        <h5>
          Test data received from http server using AJAX request
        </h5>
        <div
          class="loader"
        >
          <div
            class="loader_base"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
